File: python/THWAP/core/logging.py
import os, sys
import syslog

class thSysLogger:
	def __init__(self, ident='libTHWAP: ')
		self.ident = ident
		syslog.openlog(ident)

# LOG_EMERG, LOG_ALERT, LOG_CRIT, LOG_ERR, LOG_WARNING, LOG_NOTICE, LOG_INFO, LOG_DEBUG. 
	def logEmerg(self, msg): syslog.syslog(syslog.LOG_EMERG, msg)
	def logAlert(self, msg): syslog.syslog(syslog.LOG_ALERT, msg)
	def logCrit(self, msg): syslog.syslog(syslog.LOG_CRIT, msg)
	def logErr(self, msg): syslog.syslog(syslog.LOG_ERR, msg)
	def logWarn(self, msg): syslog.syslog(syslog.LOG_WARNING, msg)
	def logNotice(self, msg): syslog.syslog(syslog.LOG_NOTICE, msg)
	def logInfo(self, msg): syslog.syslog(syslog.LOG_INFO, msg)
	def logDebug(self, msg): syslog.syslog(syslog.LOG_DEBUG, msg)

	def cleanup(self):
		syslog.closelog()

File: ruby/core/logging.rb
require 'syslog'

class thSysLogger
	def initialize(ident)
		if ident == '': ident = $0; end
		Syslog.open(ident)
	end

	def logCrit(msg): Syslog.crit(msg); end
	def logEmerg(msg): Syslog.emerg(msg); end
	def logAlert(msg): Syslog.alert(msg); end
	def logErr(msg): Syslog.err(msg); end
	def logWarn(msg): Syslog.warn(msg); end
	def logNotice(msg): Syslog.notice(msg); end
	def logInfo(msg): Syslog.info(msg); end
	def logDebug(msg): Syslog.debug(msg); end

	def cleanup
		Syslog.close
	end
end
